### DISCUSSION TITLE
Best way to theme the editor and code mirror blocks?

### DISCUSSION THREAD

**Discussion by dave-db** [mdx-editor/editor#316] [Q&A] [answered]
I'm setting the className `${theme}-theme ${theme}-editor` where `theme` is "light" or "dark", which does a good job changing the main toolbar controls and dialogs. I'm also following the approach shown at https://github.com/mdx-editor/editor/commit/5f47d9c8ce5fe9ff7c7e1332552472274a57bb14#diff-712671d04d27ea23e100a331eb9ea09a88c109a84dd5f3f80c0afbee19eff5b9R74 to conditionally pass `basicDark` into the diffSourcePlugin when `theme === 'dark'`. It doesn't re-render if `theme` changes but it's not the end of the world.

However, the CodeMirror blocks stay light no matter what - is there a way of passing `basicDark` into the CodeMirrorPlugin as well, so they can match the styling of the main editor? The props as they are don't seem to allow it.

**Comment by petyosi** [comment, marked as answer]
That's a (necessary) missing feature. If you're in a hurry, you can just create your own codemirror editor there (or submit a PR for this one), happy to accept one.

**Comment by petyosi** [comment]
Just published a new version that exposes the theme.

**Reply by talksik** [reply]
any ideas on how to use this? Perhaps would help others too instead of leaving this be a leaky abstraction
